### PR TITLE
Switch null assignment to non-nullable entity warnings on

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

**Comment by bdach** [comment]
FYI I have a batch of changes that resolves inspections here, just want to run it past tests locally.

**Comment by bdach** [comment]
Changes are in. Because of the sheer number of the inspections being fixed the full diff is probably dreadful reviewing, but what I did instead, is I split each individual case/group of cases into separate commits, so I recommend reviewing commit-by-commit, as it'll translate to case-by-case basis. Each commit also has an explanation for the choice undertaken in the expanded description.

I attempted to avoid using `.AsNonNull()`/suppressions where possible, but there was some stuff that was too dynamic to be able to avoid using them entirely.

File: osu.Game/Skinning/Skin.cs [hunk 1]
```diff
@@ -57,7 +58,7 @@ protected Skin(SkinInfo skin, IStorageResourceProvider resources)
 
                 string jsonContent = Encoding.UTF8.GetString(bytes);
 
-                DrawableComponentInfo[skinnableTarget] = JsonConvert.DeserializeObject<IEnumerable<SkinnableInfo>>(jsonContent).ToArray();
+                DrawableComponentInfo[skinnableTarget] = JsonConvert.DeserializeObject<IEnumerable<SkinnableInfo>>(jsonContent).AsNonNull().ToArray();
```
Comment (peppy) [inline osu.Game/Skinning/Skin.cs:61]: I don't think we want to crash as a result of null here (which is why i had the checks above).

**Comment by peppy** [comment]
I think the rest seem fine.

File: osu.Game/Skinning/Skin.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Skinning/Skin.cs:61]: Fair point, tunnel-visioned a bit on that one. Figured that if a `null` falls out of json deserialisation it's a garbage-in-garbage-out scenario already, but I suppose the garbage can just as well be ignored.

Comment (bdach) [reply osu.Game/Skinning/Skin.cs]: I think this should be better now with 69fc072, if I got the intention right.

**Pull request merged**